api/src/pcapi/core/finance/api.py

```diff
@@ -869,6 +865,47 @@ def generate_cashflows(cutoff: datetime.datetime) -> models.CashflowBatch:
     return batch
 
 
+def _get_bank_accounts_for_invoices(batch: models.CashflowBatch) -> list[int]:
+    bank_account_entities = (
+        models.Pricing.query.outerjoin(models.Pricing.booking)
+        .outerjoin(bookings_models.Booking.stock)
+        .outerjoin(models.Pricing.collectiveBooking)
+        .outerjoin(educational_models.CollectiveBooking.collectiveStock)
+        .outerjoin(models.Pricing.event)
+        .outerjoin(models.Pricing.pricingPoint)
+        .outerjoin(
+            offerers_models.VenueBankAccountLink,
+            offerers_models.VenueBankAccountLink.venueId == models.Pricing.pricingPointId,
+        )
+        .outerjoin(offerers_models.VenueBankAccountLink.bankAccount)
```

Comment:
J'ai l'impression qu'on fait des `outerjoin` là où des `join` sont suffisants, car non `nullable` : à vérifier !
Ça ne change pas le résultat mais c'est plus clair.
```suggestion
        .join(models.Pricing.event)
        .join(models.Pricing.pricingPoint)
        .join(
            offerers_models.VenueBankAccountLink,
            offerers_models.VenueBankAccountLink.venueId == models.Pricing.pricingPointId,
        )
        .join(offerers_models.VenueBankAccountLink.bankAccount)
```

Reply:
tout à fait, je mets à jour 👍